java-testing/ignores/parse-for-ignores

```diff
@@ -0,0 +1,137 @@
+ignored_all() {
+  perl -pe 's/(^\s*\/\/|\s+\/\/).*$//' < "$1" \
+    | tr -d "\n" \
+    | perl -pe \
+        's/\@Ignore(\((\".*?\")\))*.*?public\s+class\s+(\S+)/\nFLAG,$2\n/g' \
+    | grep ^FLAG \
+    | perl -pe 's/FLAG/*/g'
+}
+
+###############################################################################
+# Finds ignored tests in given test file
+# Captures comment inside @Ignore(...) if present
+# Arguments:
+#   1) Java test file name
+###############################################################################
+ignored_tests() {
+ perl -pe 's/(^\s*\/\/|\s+\/\/).*$//' < "$1" \
+   | tr -d "\n" \
+   | perl -pe \
+     's/\@Ignore(\((\".*?\")\))*.*?public\s+void\s+(\S+)\(/\nFLAG$3,$2\n/g' \
+   | grep ^FLAG \
+   | perl -pe 's/FLAG//g'
+}
+
+###############################################################################
+# Parse every java file under any src/test and collect information on tests 
+# Arguments:
+#  None
+###############################################################################
+parse_test_dirs() {
+  #header for csv
+  echo -n "IGNORED/ALL IGNORED:"
+  echo -n "PROJECT,PACKAGE NAME, TEST CLASS, TEST METHOD,"
+  echo "IGNORED COMMENT, LOCAL PATH TO TEST CLASS"
+ 
+  for test_dir in ${LIST_OF_TEST_DIRS[@]}; do
+
+	echo -e "\nINFO:Running in subdir $test_dir..."
+    list_of_test_files=$(find "$test_dir" -type f -name "*.java")
+   
+    for test_file in ${list_of_test_files[@]}; do
+
+      echo "INFO:Checking test file $test_file..."
+	  test_file_name=$(basename "$test_file")
+      test_class=${test_file_name%.java}
+      package_name=$(grep "package .*;" "$test_file" \
+	  				  | awk '{print $2}' \
+	  				  | tr -d ';')
+      #Check if entire test class is ignored, if ignored continue to next 
+      all_ignored=$(ignored_all "$test_file") || true
+      if [[ -n "$all_ignored" ]]; then
+        echo -n "ALL IGNORED:$PROJECT_DIR"
+        echo ",$package_name, $test_class, $all_ignored, $test_file"
+        continue
+      fi
+
+      #Check for ignored tests in test class
+      tests_ignored=$(ignored_tests "$test_file") || true
+      if [[ -n "$tests_ignored" ]]; then       
+        while read -r line; do
+            echo -n "IGNORED:$PROJECT_DIR,"
+			echo "$package_name, $test_class, $line, $test_file"
+        done <<< "$tests_ignored"
+      fi
+	  echo "INFO:Checked test file $test_file..."
+    done
+    echo "INFO:Finished running subdir $test_dir"
+
+  done
+}
+
+###############################################################################
+# Main method
+# Arguments:
+#  1) directory to run checks in relative to dl4j repo, checks files under
+#			src/test
+###############################################################################
+main() {
+  declare PROJECT_DIR="$1"
+  if [[ -z $LIBND4J_HOME ]]; then
+    echo -n "ERROR: LIBND4J_HOME is either unset or set to a non-empty string."
+    echo -n "Set LIBND4J_HOME and rerun.Exiting script..." 1>&2
```

Comment:
Let's not use LIBND4J_HOME to locate that - we don't generally use this variable anymore, most people won't have it set.
Instead, let's assume deeplearning4j and dl4j-dev-tools are in the same parent directory
For example, deeplearning4j can be found at `../../deeplearning4j/deeplearning4j`

Reply:
k, will fix

Reply:
Does that and added an optional argument.